spark3/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreamingRead3.java

```diff
@@ -0,0 +1,502 @@
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).bucket("id", 3).build();
+    Table table = tables.create(SCHEMA, spec, location.toString());
+    final String tempView = "microBatchView";
+
+    try {
+      Dataset<Row> df = spark.readStream()
+          .format("iceberg")
+          .load(location.toString());
+      DataStreamWriter<Row> singleBatchWriter = df.writeStream()
+          .trigger(Trigger.Once())
+          .option("checkpointLocation", writerCheckpoint.toString())
+          .foreachBatch((batchDF, batchId) -> {
+            batchDF.createOrReplaceGlobalTempView(tempView);
+          });
+
+      String globalTempView = "global_temp." + tempView;
+
+      List<SimpleRecord> processStreamOnEmptyIcebergTable = processMicroBatch(singleBatchWriter, globalTempView);
+      Assert.assertEquals(
+          Collections.emptyList(),
+          processStreamOnEmptyIcebergTable);
+
+      for (List<List<SimpleRecord>> expectedCheckpoint : getTestDataForMultipleWritesWithMultipleSnapshots()) {
+        appendData(expectedCheckpoint, location);
+        table.refresh();
+
+        List<SimpleRecord> actualDataInCurrentMicroBatch = processMicroBatch(singleBatchWriter, globalTempView);
+        Assert.assertEquals(
+            expectedCheckpoint.stream().flatMap(List::stream).collect(Collectors.toList()),
+            actualDataInCurrentMicroBatch);
+      }
+    } finally {
+      for (StreamingQuery query : spark.streams().active()) {
+        query.stop();
+      }
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testParquetOrcAvroDataInOneTable() throws IOException, TimeoutException {
+    File parent = temp.newFolder("parent");
+    File location = new File(parent, "test-table");
+
+    HadoopTables tables = new HadoopTables(CONF);
+
+    // use partition buckets - so that single list<SimpleRecord> can generate multiple files per write
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).bucket("id", 3).build();
+    Table table = tables.create(SCHEMA, spec, location.toString());
+
+    List<SimpleRecord> parquetFileRecords = Lists.newArrayList(
+        new SimpleRecord(1, "one"),
+        new SimpleRecord(2, "two"),
+        new SimpleRecord(3, "three"));
+
+    List<SimpleRecord> orcFileRecords = Lists.newArrayList(
+        new SimpleRecord(4, "four"),
+        new SimpleRecord(5, "five"));
+
+    List<SimpleRecord> avroFileRecords = Lists.newArrayList(
+        new SimpleRecord(6, "six"),
+        new SimpleRecord(7, "seven"));
+
+    appendData(parquetFileRecords, location, "parquet");
+    appendData(orcFileRecords, location, "orc");
+    appendData(avroFileRecords, location, "avro");
+
+    table.refresh();
+
+
+    try {
+      Dataset<Row> ds = spark.readStream()
+          .format("iceberg")
+          .load(location.toString());
+      List<SimpleRecord> actual = processStreamTillEnd(ds);
+      List<SimpleRecord> expected = Stream.concat(Stream.concat(
```

Comment:
You may want to consider using `Iterables` instead of streams since that will work directly with the lists and is friendly with `Assertions`.

Reply:
replaced all occurences of `streams` with `iterables`.